cirq/experiments/generate_google_v2_random_circuit.py

```diff
@@ -0,0 +1,171 @@
+                                       strategy=InsertStrategy.EARLIEST)
+
+    # Add a final moment of Hadamards
+    circuit.append(common_gates.H(qubit) for qubit in qubits)
+
+    return circuit
+
+
+def generate_random_v2_circuit_grid(n_rows: int, n_cols: int,
+                                    cz_depth: int, seed: int
+                                    ) -> cirq.Circuit:
+    """
+    Generates Google Random Circuits v2 as in github.com/sboixo/GRCS cz_v2.
+    See also https://arxiv.org/abs/1807.10749
+
+    Args:
+        n_rows: number of rows of a 2D lattice.
+        n_cols: number of columns.
+        cz_depth: number of layers with CZ gates.
+        seed: seed for the random instance.
+
+    Returns:
+        A circuit corresponding to instance
+        inst_{n_rows}x{n_cols}_{cz_depth+1}_{seed}
+
+    The mapping of qubits is cirq.GridQubit(j,k) -> q[j*n_cols+k]
+    (as in the QASM mapping)
+    """
+    qubits = [cirq.GridQubit(i, j) for i in range(n_rows)
+              for j in range(n_cols)]
+    return generate_random_v2_circuit(qubits, cz_depth, seed)
+
+
+T = TypeVar('T')
+def _choice(rand_gen: Callable[[], float], sequence: Iterable[T]) -> T:
+    """
+    Choose a random element from a non-empty sequence.
+
+    Use this instead of random.choice, with random.random(), for reproducibility
+    """
+    return sequence[int(rand_gen() * len(sequence))]
+
+
+def _add_cz_layer(layer_index: int, circuit: cirq.Circuit) -> int:
+    cz_layer = None
+    while not cz_layer:
+        cz_layer = list(_make_cz_layer(circuit.all_qubits(), layer_index))
+        layer_index += 1
+
+    circuit.append(cz_layer)
+    return layer_index
+
+
+def _make_cz_layer(qubits: Iterable[GridQubit], layer_index: int
+                   ) -> Iterable[cirq.Operation]:
+    """
+    Each layer index corresponds to a shift/transpose of this CZ pattern:
+
+        ●───●   ●   ●   ●───●   ●   ● . . .
+
+        ●   ●   ●───●   ●   ●   ●───● . . .
+
+        ●───●   ●   ●   ●───●   ●   ● . . .
+
+        ●   ●   ●───●   ●   ●   ●───● . . .
+
+        ●───●   ●   ●   ●───●   ●   ● . . .
+
+        ●   ●   ●───●   ●   ●   ●───● . . .
+        .   .   .   .   .   .   .   . .
+        .   .   .   .   .   .   .   .   .
+        .   .   .   .   .   .   .   .     .
+
+    Labelled edges, showing the exact index-to-CZs mapping (mod 8):
+
+         ●─0─●─2─●─4─●─6─●─0─. . .
+        3│  7│  3│  7│  3│
+         ●─4─●─6─●─0─●─2─●─4─. . .
+        1│  5│  1│  5│  1│
+         ●─0─●─2─●─4─●─6─●─0─. . .
+        7│  3│  7│  3│  7│
+         ●─4─●─6─●─0─●─2─●─4─. . .
+        5│  1│  5│  1│  5│
+         ●─0─●─2─●─4─●─6─●─0─. . .
+        3│  7│  3│  7│  3│
+         .   .   .   .   .   .
+         .   .   .   .   .     .
+         .   .   .   .   .       .
+
+    Note that, for small devices, some layers will be empty because the layer
+    only contains edges not present on the device.
+
+    NOTE: This is the almost the function in supremacy.py,
```

Comment:
This file should supplant the existing one. Delete that one, or replace its contents with a call to this one.

Reply:
Do you mean the one in examples/supremacy.py? 
That one does not have single qubits in neighboring qubits to two qubit gates, which is sometimes useful.

Reply:
It's not too hard to put that back if needed. Unless we have an actual immediate use case for it, it should be replaced by this method.

Reply:
I think it would be better if you pull this without a deleting supremacy.py, and I also submit a pull deleting supremacy.py. Then this is easy to roll back. Is that OK?